src/Microsoft.ApiDesignGuidelines.Analyzers/Core/InterfaceMethodsShouldBeCallableByChildTypes.cs

```diff
@@ -1,39 +1,142 @@
-        
-        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
-                                                                             s_localizableTitle,
-                                                                             s_localizableMessage,
-                                                                             DiagnosticCategory.Design,
-                                                                             DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
-                                                                             description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
-                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
-
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
-
-        public override void Initialize(AnalysisContext analysisContext)
-        { 
-            
+
+        public static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                          s_localizableTitle,
+                                                                          s_localizableMessage,
+                                                                          DiagnosticCategory.Design,
+                                                                          DiagnosticSeverity.Warning,
+                                                                          isEnabledByDefault: false,
+                                                                          description: s_localizableDescription,
+                                                                          helpLinkUri: "https://msdn.microsoft.com/library/ms182153.aspx",
+                                                                          customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public sealed override void Initialize(AnalysisContext analysisContext) =>
+            analysisContext.RegisterOperationBlockAction(AnalyzeOperationBlock);
+
+        private bool ShouldExcludeOperationBlock(ImmutableArray<IOperation> operationBlocks)
+        {
+            if (operationBlocks != null && operationBlocks.Length == 1)
+            {
+                IBlockStatement block = operationBlocks[0] as IBlockStatement;
+
+                // An operation block that's not even a block - don't analyze the error cases.
+                if (block == null)
+                {
+                    return true;
+                }
+
```

Comment:
I tried working around it today but it looks its not a simple check for labelstatement. For overridden methods, there's no block at all and there's just a label for other methods there's a block and a label (sometimes inside the block, sometimes outside). Reverse engineering this bug and encoding it here convolutes this method. I'll just leave it as is.